Source/WebCore/css/SelectorChecker.cpp

```diff
@@ -763,6 +763,10 @@ bool SelectorChecker::checkOne(CheckingContext& checkingContext, LocalContext& c
         if (selector.pseudoClass() == CSSSelector::PseudoClass::Not) {
             const CSSSelectorList* selectorList = selector.selectorList();
 
+            // FIXME: We do not support combining :host with :not() functional pseudoclass. Combination with functional pseudoclass has been allowed for the useful :is(:host) ; but combining with :not() doesn't sound useful like :host():not(:not(:host))
+            if (context.mustMatchHostPseudoClass)
+                return false;
```

Comment:
nit: you can move this before `const CSSSelectorList* selectorList = selector.selectorList();`

Reply:
Does `:host :not(div)` work correctly with your patch? From my limited understanding of the code, I think this would break this case:

```
<!DOCTYPE html>
<html>
<head>
<style>
my-host {
  display: block;
  width: 100px;
  height: 100px;
  background: red;
}
</style>
</head>
<body>
<p>Test passes if you see a single 100px by 100px green box below.</p>
<my-host id="t1"></my-host>
<script>
var host = document.querySelector('#t1');
host.attachShadow({ mode: 'open' }).innerHTML = `
  <style>
    :host > div {
      background-color: red; width: 100%; height: 100%;
    }
    :host > :not(span) {
      background-color: green;
    }
  </style>
  <div></div>
`;
</script>
</body>
</html>
```

Reply:
This test case works fine, I will add this to the PR